builder/builder.go

```diff
@@ -68,9 +68,10 @@ type BuildOpt struct {
 
 // Builder executes Earthly builds.
 type Builder struct {
-	s        *solver
-	opt      Opt
-	resolver *buildcontext.Resolver
+	s         *solver
+	opt       Opt
+	resolver  *buildcontext.Resolver
+	builtMain bool
```

Comment:
Same concept as phases; just need to know one thing instead of a list index... we may need phases in future